ts/features/idpay/barcode/saga/index.ts

```diff
@@ -0,0 +1,83 @@
+import * as E from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+import { SagaIterator } from "redux-saga";
+import { call, put, takeLatest } from "typed-redux-saga/macro";
+import { ActionType } from "typesafe-actions";
+import { TransactionBarCodeResponse } from "../../../../../definitions/idpay/TransactionBarCodeResponse";
+import {
+  CodeEnum,
+  TransactionErrorDTO
+} from "../../../../../definitions/idpay/TransactionErrorDTO";
+import { SagaCallReturnType } from "../../../../types/utils";
+import { withRefreshApiCall } from "../../../fastLogin/saga/utils";
+import { IDPayClient } from "../../common/api/client";
+import { idpayGenerateBarcode } from "../store";
+
+export function* watchIDPayBarcodeSaga(
+  idPayClient: IDPayClient,
+  bearerToken: string
+): SagaIterator {
+  yield* takeLatest(
+    idpayGenerateBarcode.request,
+    handleGenerateBarcode,
+    idPayClient.createBarCodeTransaction,
+    bearerToken
+  );
+}
+
+const genericError: TransactionErrorDTO = {
+  code: CodeEnum.PAYMENT_GENERIC_ERROR,
+  message: "error"
+};
+
+export function* handleGenerateBarcode(
+  createBarCodeTransaction: IDPayClient["createBarCodeTransaction"],
+  bearerToken: string,
+  action: ActionType<typeof idpayGenerateBarcode.request>
+) {
+  const createBarCodeTransactionRequest = createBarCodeTransaction({
+    bearerAuth: bearerToken,
+    body: {
+      initiativeId: action.payload.initiativeId
+    }
+  });
+
+  try {
+    const createBarCodeTransactionResult = (yield* call(
+      withRefreshApiCall,
+      createBarCodeTransactionRequest,
+      action
+    )) as unknown as SagaCallReturnType<typeof createBarCodeTransaction>;
+
+    yield pipe(
+      createBarCodeTransactionResult,
+      E.fold(
+        _error =>
+          put(
+            idpayGenerateBarcode.failure({
+              initiativeId: action.payload.initiativeId,
+              error: genericError
+            })
+          ),
+        response =>
+          put(
+            response.status === 201
+              ? idpayGenerateBarcode.success(
+                  response.value as TransactionBarCodeResponse
+                )
+              : idpayGenerateBarcode.failure({
+                  initiativeId: action.payload.initiativeId,
+                  error: response.value
+                })
+          )
+      )
+    );
+  } catch (error) {
+    yield* put(
+      idpayGenerateBarcode.failure({
+        initiativeId: action.payload.initiativeId,
+        error: genericError
+      })
```

Comment:
I think this should be a `NetworkError`

```suggestion
      idpayGenerateBarcode.failure({
        initiativeId: action.payload.initiativeId,
        error: getNetworkError(error)
      })
```

Reply:
since all other errors here are mapped to ease error handling, I thought turning it into a readymade generic error was the way to go; I'm guessing using a `NetworkError` would add logic which could be avoided.
what do you think?

Reply:
Well, you are right, but since you added `TransactionErrorDTO | NetworkError` as the error type in the saga return type, I assumed we wanted to know when the error is a `NetworkError`.
That said, I think that knowing if the error is a `NetworkError` could come handy in the future. What do you think?

Reply:
well yes, to be honest it was a leftover from a scratched implementation 😅 
that said, it could very well come in handy, and since we'll have to map this outcome in a custom frontend error DTO it should not be much of a hassle.
got it, updating branch 👍

Reply:
addressed 👍